File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n=nums.length;
        int l=0;
        int r=n-1;
        int i=n-1;
        int[] res = new int[n];
        while(l<=r){
            if (Math.abs(nums[l])>Math.abs(nums[r])){
                res[i--]=nums[l]*nums[l];
                l+=1;
            }
            else{
                res[i--]=nums[r]*nums[r];
                r-=1;
            }
        }
        return res;
    }
}

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        l=0
        r=len(nums)-1
        temp=[]
        while l<=r:
            if (nums[l])**2>nums[r]**2:
                temp.append(nums[l]**2)
                l+=1
            else:
                temp.append(nums[r]**2)
                r-=1
        return temp[::-1]
                
        